HIP/hip-583.md

```diff
@@ -2,221 +2,268 @@
+`alias`.
 
 ## Rationale
 
-With [HIP 32](https://hips.hedera.com/hip/hip-32) it’s possible to send HBARs in a `CryptoTransfer` to an account before it exists on the ledger by providing an alias using a cryptographic public key (`ED25519` or `ECDSA`) of the `public-key` format. 
-Using this, the ledger will confirm the uniqueness of the alias and create an account that can be referenced via this accounts public key alias.
-
-However, the ability to create accounts or refer to accounts with an alias is limited and confusing for Smart Contract user experiences since
-
-- Aliases can only be set using a `CryptoTransfer` transaction, whereas wallets would prefer to do this in a `CryptoCreate` transaction in addition to a `CrytoTransfer` transaction.
-- Users who only know their `public-address` must find innovative ways to find their public key or safely export their private key to extract the public key for use in the `CryptoTransfer`.
-- HBAR or token transfers using an alias only support the `public-key` alias format and not the `public-address` format.
-
-Notably, most ledgers in the market utilize the `ECDSA` key for accounts, especially where an `EVM` is concerned. In fact to completely support all Smart Contract scenarios on the `EVM` used by Hedera, an account must possess an alias based on its `ECDSA` public key that dictates the address value observed by teh EVM in transactuions. 
+Fundamental in the design of the Hedera network are the concepts of _accounts_, account _identifiers_, and _keys_. These
+are fundamentally different entities. An `Account` must have a key. The key for the account may be a simple key
+(typically, an `ED25519` public key), or it may be a compound key, such as a list of keys all of which must sign each
+transaction, or a threshold key such that N of the keys must be signed. These may be nested to create very rich and
+powerful accounts requiring multiple-signatures. Further, an `Account` may change the keys on the account over time.
+
+An `Account` also has a system-assigned account identifier of the form `shard.realm.num`, where each value is an 8-byte
+number. It may also be referred to with an identifier of the form `shard.realm.alias`, where the alias is of one of
+the following forms:
+
+1. `long-zero alias` - This is the Hedera account number styled as an Ethereum account hex string. The
+   Hedera account number is hex encoded and the value is prefixed with zeros to form a 20 byte address. This format
+   cannot be used to auto-create an account, but can be used wherever an `alias` can be used.
+2. `key alias` - This format is created by using the bytes of a simple cryptographic public key supported by Hedera
+   (such as `ED25519` or `ECDSA secp256k1`). It is not necessary for the account's key to match the `alias`, only that
+   the account owner has asserted cryptographic proof of ownership of the account `alias` key.
```

Comment:
Q: Should we add that on creation if if a key alias is provided that it will match the account key, or are we providing anew feature here and relaxing this rule? If so it'll mean you can create an account with key A and then key alias B in one go

Reply:
To be clear I think in the case where account key is A and key alias is B, we'd preserve existing network logic that checks that the account id key is present in the signature map and for alias the key is also present in the signature map.

Reply:
Yes exactly.

Reply:
Just to make sure we do not miss something:

The current signing requirements for `CryptoCreate` [only require a signature from the new account key, i.e. the `adminКey`, when the `receiverSigRequried` is being set to `true`.](https://github.com/hashgraph/hedera-services/blob/8d6869569f5874f3ad033907f417b0a7fb363fe6/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java#L775-L785) Otherwise, no signature is needed (except for the payer's, of course).

Are we aiming to change this?